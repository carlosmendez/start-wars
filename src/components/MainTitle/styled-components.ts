import styled from '@emotion/styled';

export const StyledMainTitle = styled.div`
  text-align: center;
  h2 {
    color: rgb(255, 228, 0);
  }
`;
