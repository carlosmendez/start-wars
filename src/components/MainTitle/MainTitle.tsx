import React, { ReactNode } from 'react';
import { StyledMainTitle } from "./styled-components";

type MainTitleProps = {
  name: string;
};

const MainTitle = ({
  name,
}: MainTitleProps) => {
  return (
    <StyledMainTitle>
      {name}
    </StyledMainTitle>
  )
}

export { MainTitle };
