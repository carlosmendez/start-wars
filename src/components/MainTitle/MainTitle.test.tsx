import React from 'react';
import renderer from 'react-test-renderer';
import { MainTitle } from "./MainTitle";

const props = {
  name: "Star Wars characters"
};

describe('MainTitle component', () => {
  it('renders correctly', () => {
    const jsx = (
      <MainTitle {...props} />
    );
    const tree = renderer
      .create(jsx)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
