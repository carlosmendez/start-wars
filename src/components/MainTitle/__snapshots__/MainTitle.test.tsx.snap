// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<h1
  className="css-1yltisf"
>
  Star Wars characters
</h1>
`;

exports[`MainTitle component renders correctly 1`] = `
<h1
  className="css-1yltisf"
>
  Star Wars characters
</h1>
`;
